src/unix/os390-syscalls.c

```diff
@@ -63,18 +63,15 @@ int scandir(const char* maindir, struct dirent*** namelist,
       struct dirent* copy;
       copy = uv__malloc(sizeof(*copy));
       if (!copy) {
-        while (count) {
-          dirent = nl[--count];
-          uv__free(dirent);
-        }
-        uv__free(nl);
-        closedir(mdir);
-        errno = ENOMEM;
-        return -1;
+        goto error;
       }
       memcpy(copy, dirent, sizeof(*copy));
 
       nl = uv__realloc(nl, sizeof(*copy) * (count + 1));
+      if (nl == NULL) {
+        uv__free(copy);
```

Comment:
This is just me thinking aloud but it might be good to introduce `uv__reallocf()` (inspired by freebsd's [reallocf(3)](https://www.freebsd.org/cgi/man.cgi?query=reallocf)) that frees the memory on realloc failure.

Reply:
Sound like a good idea as `realloc` nullptr check is present in quite a few places.

Reply:
Aw, I realized just after merging that this has a bug: `nl == NULL` now but it's indexed on line 87. I'll open a PR...